iceoryx2/src/service/static_config/message_type_details.rs

```diff
@@ -20,9 +20,10 @@ use serde::{Deserialize, Serialize};
 #[derive(Default, Debug, Clone, Eq, Hash, PartialEq, Serialize, Deserialize)]
 pub enum TypeVariant {
     #[default]
-    /// A fixed size type like [`u64`]
+    /// A fixed size type which is determined during compile.
+    /// A structure holds some pointer fields should be clarified as Dynamic.
```

Comment:
Oh this would be wrong. With dynamic we mean a slice `[MY_TYPE]` nothing else. 

Yesterday, someone already run into the issue that our payload type restrictions was not clearly documented, see #436 . 
So, any payload type that you want to transmit with iceoryx2 must be:
 * self-contained, no heap is allowed
 * cannot contain pointer members
 * must be annotated with `#[repr(C)]`

The underlying reason is the shared memory which we use to store the payload data. If the payload type would use the heap then the type would use process local memory that is not available to another process.

The pointer requirement comes again from shared memory. It has a different pointer address offset in every process rendering any absolut pointer useless and dereferencing it would end up in a segfault.

The `#[repr(C)]` requirement prevents the rust compiler of reordering the members of a struct. Just think of that two processes use the same struct but have a different member order, this would result again in undefined behavior.

And dynamic means that you can use a slice of iceoryx2 compatible types.